Eigen error when starting

From what I have been reading my sensor should not work with the foxy-devel branch but it does although it flickers a bit on the left side of the sensor. 

To check if this was fixed in the newer versions I tried to run the galactic version and the version in the ros2 branch (on a machine with galactic installed) but both gave me the same error:

```
[INFO] [launch]: All log files can be found below /home/bertaveira/.ros/log/2022-01-19-21-57-23-674269-ubuntu-7377
[INFO] [launch]: Default logging verbosity is set to INFO
1642625843.766833 [0]       ros2: using network interface enp0s5 (udp/192.168.0.110) selected arbitrarily from: enp0s5, docker0, enxa0cec8ea264f
[INFO] [ouster_driver-1]: process started with pid [7388]
[ouster_driver-1] 1642625843.840869 [0] ouster_dri: using network interface enp0s5 (udp/192.168.0.110) selected arbitrarily from: enp0s5, docker0, enxa0cec8ea264f
[ouster_driver-1] [INFO] [1642625844.036431420] [ouster_driver]: Configuring Ouster driver node.
[ouster_driver-1] [INFO] [1642625844.037142796] [ouster_driver]: This driver is compatible with sensors running fw 2.x.
[ouster_driver-1] [INFO] [1642625844.037468255] [ouster_driver]: Connecting to sensor at 10.5.5.66.
[ouster_driver-1] [INFO] [1642625844.037495880] [ouster_driver]: Sending data from sensor to 10.5.5.1.
[INFO] [launch.user]: [LifecycleLaunch] Ouster driver node is activating.
[ouster_driver-1] [INFO] [1642625861.387698393] [ouster_driver]: Activating Ouster driver node.
[ouster_driver-1] ouster_driver: /usr/include/eigen3/Eigen/src/Core/DenseCoeffsBase.h:180: Eigen::DenseCoeffsBase<Derived, 0>::CoeffReturnType Eigen::DenseCoeffsBase<Derived, 0>::operator()(Eigen::Index) const [with Derived = Eigen::Map<const Eigen::Array<unsigned int, -1, -1, 1, -1, -1> >; Eigen::DenseCoeffsBase<Derived, 0>::CoeffReturnType = unsigned int; Eigen::Index = long int]: Assertion `index >= 0 && index < size()' failed.
[ERROR] [ouster_driver-1]: process has died [pid 7388, exit code -6, cmd '/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/ros2_ouster/ouster_driver --ros-args -r __node:=ouster_driver -r __ns:=/ --params-file /media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/share/ros2_ouster/params/sensor.yaml'].
[INFO] [launch.user]: [LifecycleLaunch] Ouster driver node is exiting.
```
Sensor: OS1-64
Firmware: v2.1.3
OS: Ubuntu 20.04
ROS: foxy and Galactic

Ran it in debug mode and here is the output from beginning to the crash:
```
ros2 launch ros2_ouster os1_launch.py 
[INFO] [launch]: All log files can be found below /home/bertaveira/.ros/log/2022-01-20-20-08-53-658430-ubuntu-221276
[INFO] [launch]: Default logging verbosity is set to INFO
1642705733.776111 [0]       ros2: using network interface enp0s5 (udp/192.168.0.110) selected arbitrarily from: enp0s5, docker0, enxa0cec8ea264f
[INFO] [ouster_driver-1]: process started with pid [221287]
[ouster_driver-1] [DEBUG] [1642705733.886581062] [rclcpp]: signal handler installed
[ouster_driver-1] [DEBUG] [1642705733.886609520] [rclcpp]: deferred_signal_handler(): waiting for SIGINT or uninstall
[ouster_driver-1] [DEBUG] [1642705733.888535313] [rcl]: Initializing node 'OusterDriver' in namespace ''
[ouster_driver-1] [DEBUG] [1642705733.888553521] [rcl]: Using domain ID of '0'
[ouster_driver-1] 1642705733.888935 [0] ouster_dri: using network interface enp0s5 (udp/192.168.0.110) selected arbitrarily from: enp0s5, docker0, enxa0cec8ea264f
[ouster_driver-1] [DEBUG] [1642705733.891150355] [rcl]: Initializing publisher for topic name '/rosout'
[ouster_driver-1] [DEBUG] [1642705733.891198480] [rcl]: Expanded topic name '/rosout'
[ouster_driver-1] [DEBUG] [1642705733.892353939] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705733.892367772] [rcl]: Node initialized
[ouster_driver-1] [DEBUG] [1642705733.894800940] [rcl]: Initializing service for service name 'ouster_driver/get_parameters'
[ouster_driver-1] [DEBUG] [1642705733.894815815] [rcl]: Expanded service name '/ouster_driver/get_parameters'
[ouster_driver-1] [DEBUG] [1642705733.895987940] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.896009940] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_parametersRequest
[ouster_driver-1] [DEBUG] [1642705733.896012398] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_parametersReply
[ouster_driver-1] [DEBUG] [1642705733.896038773] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.896145732] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.896179857] [rcl]: Initializing service for service name 'ouster_driver/get_parameter_types'
[ouster_driver-1] [DEBUG] [1642705733.896187065] [rcl]: Expanded service name '/ouster_driver/get_parameter_types'
[ouster_driver-1] [DEBUG] [1642705733.896489232] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.896500648] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_parameter_typesRequest
[ouster_driver-1] [DEBUG] [1642705733.896503273] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_parameter_typesReply
[ouster_driver-1] [DEBUG] [1642705733.896508065] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.896597982] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.896620898] [rcl]: Initializing service for service name 'ouster_driver/set_parameters'
[ouster_driver-1] [DEBUG] [1642705733.896625065] [rcl]: Expanded service name '/ouster_driver/set_parameters'
[ouster_driver-1] [DEBUG] [1642705733.896941565] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.896952398] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/set_parametersRequest
[ouster_driver-1] [DEBUG] [1642705733.896959440] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/set_parametersReply
[ouster_driver-1] [DEBUG] [1642705733.896963523] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.897049815] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.897065940] [rcl]: Initializing service for service name 'ouster_driver/set_parameters_atomically'
[ouster_driver-1] [DEBUG] [1642705733.897083774] [rcl]: Expanded service name '/ouster_driver/set_parameters_atomically'
[ouster_driver-1] [DEBUG] [1642705733.897801774] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.897813732] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/set_parameters_atomicallyRequest
[ouster_driver-1] [DEBUG] [1642705733.897816524] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/set_parameters_atomicallyReply
[ouster_driver-1] [DEBUG] [1642705733.897822107] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.897919857] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.897934940] [rcl]: Initializing service for service name 'ouster_driver/describe_parameters'
[ouster_driver-1] [DEBUG] [1642705733.897941190] [rcl]: Expanded service name '/ouster_driver/describe_parameters'
[ouster_driver-1] [DEBUG] [1642705733.898335232] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.898344941] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/describe_parametersRequest
[ouster_driver-1] [DEBUG] [1642705733.898347107] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/describe_parametersReply
[ouster_driver-1] [DEBUG] [1642705733.898352649] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.898430191] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.898467732] [rcl]: Initializing service for service name 'ouster_driver/list_parameters'
[ouster_driver-1] [DEBUG] [1642705733.898474774] [rcl]: Expanded service name '/ouster_driver/list_parameters'
[ouster_driver-1] [DEBUG] [1642705733.898984816] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.898995857] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/list_parametersRequest
[ouster_driver-1] [DEBUG] [1642705733.899046149] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/list_parametersReply
[ouster_driver-1] [DEBUG] [1642705733.899057649] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.899185274] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.899213066] [rcl]: Initializing publisher for topic name '/parameter_events'
[ouster_driver-1] [DEBUG] [1642705733.899223066] [rcl]: Expanded topic name '/parameter_events'
[ouster_driver-1] [DEBUG] [1642705733.899863566] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705733.902780358] [rcl]: Initializing subscription for topic name '/parameter_events'
[ouster_driver-1] [DEBUG] [1642705733.902803108] [rcl]: Expanded topic name '/parameter_events'
[ouster_driver-1] [DEBUG] [1642705733.903032359] [rcl]: Subscription initialized
[ouster_driver-1] [DEBUG] [1642705733.903681650] [rcl]: Initializing publisher for topic name '~/transition_event'
[ouster_driver-1] [DEBUG] [1642705733.903692734] [rcl]: Expanded topic name '/ouster_driver/transition_event'
[ouster_driver-1] [DEBUG] [1642705733.904408984] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705733.904427817] [rcl]: Initializing service for service name '~/change_state'
[ouster_driver-1] [DEBUG] [1642705733.904434776] [rcl]: Expanded service name '/ouster_driver/change_state'
[ouster_driver-1] [DEBUG] [1642705733.905012651] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.905025359] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/change_stateRequest
[ouster_driver-1] [DEBUG] [1642705733.905027567] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/change_stateReply
[ouster_driver-1] [DEBUG] [1642705733.905035067] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.905235859] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.905245359] [rcl]: Initializing service for service name '~/get_state'
[ouster_driver-1] [DEBUG] [1642705733.905248943] [rcl]: Expanded service name '/ouster_driver/get_state'
[ouster_driver-1] [DEBUG] [1642705733.905584068] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.905594609] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_stateRequest
[ouster_driver-1] [DEBUG] [1642705733.905599818] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_stateReply
[ouster_driver-1] [DEBUG] [1642705733.905604401] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.905703984] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.905711568] [rcl]: Initializing service for service name '~/get_available_states'
[ouster_driver-1] [DEBUG] [1642705733.905715026] [rcl]: Expanded service name '/ouster_driver/get_available_states'
[ouster_driver-1] [DEBUG] [1642705733.906022276] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.906033026] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_available_statesRequest
[ouster_driver-1] [DEBUG] [1642705733.906035776] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_available_statesReply
[ouster_driver-1] [DEBUG] [1642705733.906042776] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.906173234] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.906185026] [rcl]: Initializing service for service name '~/get_available_transitions'
[ouster_driver-1] [DEBUG] [1642705733.906192068] [rcl]: Expanded service name '/ouster_driver/get_available_transitions'
[ouster_driver-1] [DEBUG] [1642705733.906575193] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.906584901] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_available_transitionsRequest
[ouster_driver-1] [DEBUG] [1642705733.906587110] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_available_transitionsReply
[ouster_driver-1] [DEBUG] [1642705733.906592568] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.906773651] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.906784693] [rcl]: Initializing service for service name '~/get_transition_graph'
[ouster_driver-1] [DEBUG] [1642705733.906787818] [rcl]: Expanded service name '/ouster_driver/get_transition_graph'
[ouster_driver-1] [DEBUG] [1642705733.906806526] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705733.906815901] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_transition_graphRequest
[ouster_driver-1] [DEBUG] [1642705733.906818151] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_transition_graphReply
[ouster_driver-1] [DEBUG] [1642705733.906820276] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705733.906912401] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705733.907756818] [rcl]: Initializing wait set with '0' subscriptions, '2' guard conditions, '0' timers, '0' clients, '0' services
[ouster_driver-1] [DEBUG] [1642705733.907852110] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.907860777] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.907890860] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.907896402] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.907898360] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.907913777] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.907930068] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908000152] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908007110] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908014943] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908016818] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908020318] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908025402] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908031235] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908036985] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908039777] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908041610] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908045985] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908049152] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908052777] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908054735] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908057235] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908060943] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908063568] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908069193] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908072860] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908077693] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908080277] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908084402] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908087193] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908091985] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908098610] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908102943] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908107068] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908108943] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908111818] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908117235] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908120902] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908125318] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908131818] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908133902] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908136568] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908139818] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908248152] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908258068] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908263027] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908267902] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908272360] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908279568] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908287235] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908291152] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908293860] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908298902] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908301818] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908312360] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908320402] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908328235] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908331152] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908335235] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908338193] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908344360] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908349068] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908353735] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908356443] [rcl]: Subscription in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908359902] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908362652] [rcl]: Subscription taking message
[ouster_driver-1] [DEBUG] [1642705733.908368193] [rcl]: Subscription take succeeded: true
[ouster_driver-1] [DEBUG] [1642705733.908371902] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908376610] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705733.908379277] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705733.908382652] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705733.908387360] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705734.056029736] [rcl]: Service in wait set is ready
[ouster_driver-1] [DEBUG] [1642705734.056086153] [rcl]: Service server taking service request
[ouster_driver-1] [DEBUG] [1642705734.056099569] [rcl]: Service take request succeeded: true
[ouster_driver-1] [INFO] [1642705734.056181361] [ouster_driver]: Configuring Ouster driver node.
[ouster_driver-1] [INFO] [1642705734.056586111] [ouster_driver]: This driver is compatible with sensors running fw 2.x.
[ouster_driver-1] [INFO] [1642705734.056869403] [ouster_driver]: Connecting to sensor at 10.5.5.66.
[ouster_driver-1] [INFO] [1642705734.056898653] [ouster_driver]: Sending data from sensor to 10.5.5.1.
[ouster_driver-1] [DEBUG] [1642705734.059143112] [rcl]: Initializing service for service name '~/reset'
[ouster_driver-1] [DEBUG] [1642705734.059164612] [rcl]: Expanded service name '/ouster_driver/reset'
[ouster_driver-1] [DEBUG] [1642705734.059951571] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705734.059967112] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/resetRequest
[ouster_driver-1] [DEBUG] [1642705734.059969904] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/resetReply
[ouster_driver-1] [DEBUG] [1642705734.059984612] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705734.060176987] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705734.060707279] [rcl]: Initializing service for service name '~/get_metadata'
[ouster_driver-1] [DEBUG] [1642705734.060721737] [rcl]: Expanded service name '/ouster_driver/get_metadata'
[ouster_driver-1] [DEBUG] [1642705734.062106196] [rmw_cyclonedds_cpp]: ************ Service Details *********
[ouster_driver-1] [DEBUG] [1642705734.062120946] [rmw_cyclonedds_cpp]: Sub Topic rq/ouster_driver/get_metadataRequest
[ouster_driver-1] [DEBUG] [1642705734.062123571] [rmw_cyclonedds_cpp]: Pub Topic rr/ouster_driver/get_metadataReply
[ouster_driver-1] [DEBUG] [1642705734.062129613] [rmw_cyclonedds_cpp]: ***********
[ouster_driver-1] [DEBUG] [1642705734.062281780] [rcl]: Service initialized
[ouster_driver-1] [DEBUG] [1642705751.065548374] [rcl]: Initializing publisher for topic name 'range_image'
[ouster_driver-1] [DEBUG] [1642705751.065705333] [rcl]: Expanded topic name '/range_image'
[ouster_driver-1] [DEBUG] [1642705751.073797502] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.076015336] [rcl]: Initializing publisher for topic name 'intensity_image'
[ouster_driver-1] [DEBUG] [1642705751.076071003] [rcl]: Expanded topic name '/intensity_image'
[ouster_driver-1] [DEBUG] [1642705751.076353586] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.076413419] [rcl]: Initializing publisher for topic name 'ambient_image'
[ouster_driver-1] [DEBUG] [1642705751.076426003] [rcl]: Expanded topic name '/ambient_image'
[ouster_driver-1] [DEBUG] [1642705751.076636545] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.121089976] [rcl]: Initializing publisher for topic name 'points'
[ouster_driver-1] [DEBUG] [1642705751.121142601] [rcl]: Expanded topic name '/points'
[ouster_driver-1] [DEBUG] [1642705751.122220518] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.122786268] [rcl]: Initializing publisher for topic name 'imu'
[ouster_driver-1] [DEBUG] [1642705751.122807352] [rcl]: Expanded topic name '/imu'
[ouster_driver-1] [DEBUG] [1642705751.123410644] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.123897435] [rcl]: Initializing publisher for topic name 'scan'
[ouster_driver-1] [DEBUG] [1642705751.123916352] [rcl]: Expanded topic name '/scan'
[ouster_driver-1] [DEBUG] [1642705751.124417644] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.125024936] [rcl]: Initializing publisher for topic name '/tf_static'
[ouster_driver-1] [DEBUG] [1642705751.125057311] [rcl]: Expanded topic name '/tf_static'
[ouster_driver-1] [DEBUG] [1642705751.126110769] [rcl]: Publisher initialized
[ouster_driver-1] [DEBUG] [1642705751.126795436] [rcl]: Sending service response
[INFO] [launch.user]: [LifecycleLaunch] Ouster driver node is activating.
[ouster_driver-1] [DEBUG] [1642705751.127404978] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705751.127423603] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705751.127451937] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.127463937] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.127476353] [rcl]: Waiting without timeout
[ouster_driver-1] [DEBUG] [1642705751.127492395] [rcl]: Timeout calculated based on next scheduled timer: false
[ouster_driver-1] [DEBUG] [1642705751.129177645] [rcl]: Service in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.129197020] [rcl]: Service server taking service request
[ouster_driver-1] [DEBUG] [1642705751.129204187] [rcl]: Service take request succeeded: true
[ouster_driver-1] [INFO] [1642705751.129236479] [ouster_driver]: Activating Ouster driver node.
[ouster_driver-1] [DEBUG] [1642705751.129457687] [rcl]: Initializing timer with period: 390625ns
[ouster_driver-1] [DEBUG] [1642705751.129528146] [rcl]: Sending service response
[ouster_driver-1] [DEBUG] [1642705751.129571021] [rcl]: Waiting with timeout: 0s + 290292ns
[ouster_driver-1] [DEBUG] [1642705751.129577771] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.129592521] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.129598479] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.129603271] [rcl]: Waiting with timeout: 0s + 258042ns
[ouster_driver-1] [DEBUG] [1642705751.129607104] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.129935021] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.129997437] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.130247104] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.132798647] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.132813813] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.132822813] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.132828313] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.132832688] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.132854230] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.133293147] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.133304522] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.133308230] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.133313730] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.133321230] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.133328272] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.133806272] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.133823855] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.133832272] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.133834772] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.133846439] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.133864855] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.134313939] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.134327231] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.134331231] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.134333147] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.134336731] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.134342564] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.134815939] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.134829356] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.134835731] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.134876356] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.134887897] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.134911939] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.135329648] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.135339648] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.135343523] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.135349148] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.135352356] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.135391773] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.135809439] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.135819023] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.135823773] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.135825606] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.135828481] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.135840356] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.136229398] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.136238773] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.136241981] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.136243815] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.136250940] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.136261398] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.136648106] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.136656856] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.136659940] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.136661815] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.136664440] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.136678648] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.137067440] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.137076273] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.137079273] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.137081065] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.137083565] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.137093981] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.137487773] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.137496690] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.137499773] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.137501648] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.137510648] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.137519315] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.137914315] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.137926023] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.137929107] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.137930940] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.137933523] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.137946857] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.138336899] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.138350649] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.138353607] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.138355357] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.138357857] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.138361649] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.138759482] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.138775232] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.138778232] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.138779982] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.138804815] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.138822107] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.139200357] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.139215649] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.139218816] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.139220607] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.139223066] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.139226857] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.139613316] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.139625941] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.139629191] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.139630941] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.139633441] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.139665441] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.140048441] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.140063566] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.140066732] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.140107607] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.140125316] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.140156441] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.140608608] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.140615733] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.140622483] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.140624483] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.140627399] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.140632358] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.141062233] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.141070483] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.141074358] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.141076233] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.141078983] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.141083441] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.141472191] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.141478566] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.141481900] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.141483941] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.141486650] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.141490733] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.141866275] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.141871983] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.141875150] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.141877150] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.141879733] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.141883608] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] [DEBUG] [1642705751.142313775] [rcl]: Waiting with timeout: 0s + 0ns
[ouster_driver-1] [DEBUG] [1642705751.142321817] [rcl]: Timeout calculated based on next scheduled timer: true
[ouster_driver-1] [DEBUG] [1642705751.142325400] [rcl]: Timer in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.142327400] [rcl]: Guard condition in wait set is ready
[ouster_driver-1] [DEBUG] [1642705751.142329983] [rcl]: Calling timer
[ouster_driver-1] [DEBUG] [1642705751.142334817] [ouster_driver]: Packet with state: imu data lidar data 
[ouster_driver-1] ouster_driver: /usr/include/eigen3/Eigen/src/Core/DenseCoeffsBase.h:180: Eigen::DenseCoeffsBase<Derived, 0>::CoeffReturnType Eigen::DenseCoeffsBase<Derived, 0>::operator()(Eigen::Index) const [with Derived = Eigen::Map<const Eigen::Array<unsigned int, -1, -1, 1, -1, -1> >; Eigen::DenseCoeffsBase<Derived, 0>::CoeffReturnType = unsigned int; Eigen::Index = long int]: Assertion `index >= 0 && index < size()' failed.
[ERROR] [ouster_driver-1]: process has died [pid 221287, exit code -6, cmd '/home/bertaveira/Desktop/Parallels Shared Folders/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/ros2_ouster/ouster_driver --ros-args --log-level debug --ros-args -r __node:=ouster_driver -r __ns:=/ --params-file /home/bertaveira/Desktop/Parallels Shared Folders/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/share/ros2_ouster/params/sensor.yaml'].
[INFO] [launch.user]: [LifecycleLaunch] Ouster driver node is exiting.

```

@SteveMacenski Thanks. I am not the most knowledgable person to solve the issue but I will try to at least provide as much information as possible. I don't have a machine with a clean galactic installed so thats why I was trying docker. All my tests were indeed with fresh docker container with only those two dependencies. 

I will try installing the dependencies via rosdep instead of manual installing those two to check if the same persists. 

In the meantime I ran it with backward_ros and got a stack trace:
```
[ouster_driver-1] ouster_driver: /usr/include/eigen3/Eigen/src/Core/DenseCoeffsBase.h:180: Eigen::DenseCoeffsBase<Derived, 0>::CoeffReturnType Eigen::DenseCoeffsBase<Derived, 0>::operator()(Eigen::Index) const [with Derived = Eigen::Map<const Eigen::Array<unsigned int, -1, -1, 1, -1, -1> >; Eigen::DenseCoeffsBase<Derived, 0>::CoeffReturnType = unsigned int; Eigen::Index = long int]: Assertion `index >= 0 && index < size()' failed.
[ouster_driver-1] Stack trace (most recent call last):
[ouster_driver-1] #19   Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/ros2_ouster/ouster_driver", at 0xaaaace8379f3, in $x
[ouster_driver-1] #18   Source "../csu/libc-start.c", line 308, in __libc_start_main [0xffff8674814f]
[ouster_driver-1] #17   Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/ros2_ouster/ouster_driver", at 0xaaaace837ffb, in main
[ouster_driver-1] #16   Object "/opt/ros/foxy/lib/librclcpp.so", at 0xffff86b91317, in rclcpp::spin(std::shared_ptr<rclcpp::node_interfaces::NodeBaseInterface>)
[ouster_driver-1] #15   Object "/opt/ros/foxy/lib/librclcpp.so", at 0xffff86b9459b, in rclcpp::executors::SingleThreadedExecutor::spin()
[ouster_driver-1] #14   Object "/opt/ros/foxy/lib/librclcpp.so", at 0xffff86b8ea13, in rclcpp::Executor::execute_any_executable(rclcpp::AnyExecutable&)
[ouster_driver-1] #13   Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff8704ee77, in rclcpp::GenericTimer<std::_Bind<void (ros2_ouster::OusterDriver::*(ros2_ouster::OusterDriver*))()>, (void*)0>::execute_callback()
[ouster_driver-1] #12   Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff87051207, in void rclcpp::GenericTimer<std::_Bind<void (ros2_ouster::OusterDriver::*(ros2_ouster::OusterDriver*))()>, (void*)0>::execute_callback_delegate<std::_Bind<void (ros2_ouster::OusterDriver::*(ros2_ouster::OusterDriver*))()>, (void*)0>()
[ouster_driver-1] #11   Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff87052c87, in void std::_Bind<void (ros2_ouster::OusterDriver::*(ros2_ouster::OusterDriver*))()>::operator()<, void>()
[ouster_driver-1] #10   Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff870542bf, in void std::_Bind<void (ros2_ouster::OusterDriver::*(ros2_ouster::OusterDriver*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>)
[ouster_driver-1] #9    Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff87054e8f, in std::__invoke_result<void (ros2_ouster::OusterDriver::*&)(), ros2_ouster::OusterDriver*&>::type std::__invoke<void (ros2_ouster::OusterDriver::*&)(), ros2_ouster::OusterDriver*&>(void (ros2_ouster::OusterDriver::*&)(), ros2_ouster::OusterDriver*&)
[ouster_driver-1] #8    Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff8705584b, in void std::__invoke_impl<void, void (ros2_ouster::OusterDriver::*&)(), ros2_ouster::OusterDriver*&>(std::__invoke_memfun_deref, void (ros2_ouster::OusterDriver::*&)(), ros2_ouster::OusterDriver*&)
[ouster_driver-1] #7    Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff86fc0dd3, in ros2_ouster::OusterDriver::processData()
[ouster_driver-1] #6    Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff86fc9a7f, in sensor::ScanProcessor::process(unsigned char const*, unsigned long)
[ouster_driver-1] #5    Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff86fc5aef, in ros2_ouster::toMsg(ouster::LidarScan, std::chrono::duration<long, std::ratio<1l, 1000000000l> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ouster::sensor::sensor_info const&, unsigned char, unsigned long)
[ouster_driver-1] #4    Object "/media/psf/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/libouster_driver_core.so", at 0xffff86fcd5e7, in Eigen::DenseCoeffsBase<Eigen::Map<Eigen::Array<unsigned int, -1, -1, 1, -1, -1> const, 0, Eigen::Stride<0, 0> >, 0>::operator()(long) const
[ouster_driver-1] #3    Source "/build/glibc-vWotZY/glibc-2.31/assert/assert.c", line 101, in __assert_fail [0xffff867547a3]
[ouster_driver-1] #2    Source "/build/glibc-vWotZY/glibc-2.31/assert/assert.c", line 92, in __assert_fail_base [0xffff8675473f]
[ouster_driver-1] #1    Source "/build/glibc-vWotZY/glibc-2.31/stdlib/abort.c", line 79, in abort [0xffff86747deb]
[ouster_driver-1] #0    Source "../sysdeps/unix/sysv/linux/raise.c", line 51, in raise [0xffff8675b278]
[ouster_driver-1] Aborted (Signal sent by tkill() 6047 1000)
[ERROR] [ouster_driver-1]: process has died [pid 6047, exit code -6, cmd '/home/bertaveira/Desktop/Parallels Shared Folders/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/lib/ros2_ouster/ouster_driver --ros-args --log-level debug --ros-args -r __node:=ouster_driver -r __ns:=/ --params-file /home/bertaveira/Desktop/Parallels Shared Folders/Home/Documents/CFS/dv/ROS_packages/ouster-drivers/install/ros2_ouster/share/ros2_ouster/params/sensor.yaml'].
[INFO] [launch.user]: [LifecycleLaunch] Ouster driver node is exiting.
```

--- a/ros2_ouster/include/ros2_ouster/conversions.hpp
+++ b/ros2_ouster/include/ros2_ouster/conversions.hpp
@@ -231,6 +231,7 @@ inline sensor_msgs::msg::LaserScan toMsg(
     msg.intensities.push_back(
       static_cast<float>((ls.field(ouster::LidarScan::INTENSITY)(i)))
     );
+    if (i == 0) break; // Fix #90
   }
 
   return msg;

--- a/ros2_ouster/include/ros2_ouster/conversions.hpp
+++ b/ros2_ouster/include/ros2_ouster/conversions.hpp
@@ -224,6 +224,9 @@ inline sensor_msgs::msg::LaserScan toMsg(
     ouster::sensor::n_cols_of_lidar_mode(mdata.mode);
   msg.angle_increment = 2 * M_PI / ouster::sensor::n_cols_of_lidar_mode(mdata.mode);
 
+  // Fix #90 (PR #92) - The iterator is in the loop condition to prevent overflow by
+  // decrementing unsigned variable 'i' bellow 0. This happened when ring 0 was selected
+  // due to the condition being reduced to i >= 0
   for (size_t i = ls.w * ring_to_use + ls.w; i-- > ls.w * ring_to_use;) {
     msg.ranges.push_back(
       static_cast<float>((ls.field(ouster::LidarScan::RANGE)(i) * ouster::sensor::range_unit))

--- a/ros2_ouster/include/ros2_ouster/conversions.hpp
+++ b/ros2_ouster/include/ros2_ouster/conversions.hpp
@@ -224,14 +224,13 @@ inline sensor_msgs::msg::LaserScan toMsg(
     ouster::sensor::n_cols_of_lidar_mode(mdata.mode);
   msg.angle_increment = 2 * M_PI / ouster::sensor::n_cols_of_lidar_mode(mdata.mode);
 
-  for (size_t i = ls.w * ring_to_use + ls.w - 1; i >= ls.w * ring_to_use; i--) {
+  for (size_t i = ls.w * ring_to_use + ls.w; i-- > ls.w * ring_to_use;) {
     msg.ranges.push_back(
       static_cast<float>((ls.field(ouster::LidarScan::RANGE)(i) * ouster::sensor::range_unit))
     );
     msg.intensities.push_back(
       static_cast<float>((ls.field(ouster::LidarScan::INTENSITY)(i)))
     );
-    if (i == 0) break; // Fix #90
   }
 
   return msg;

